adhocracy4/categories/assets/category_formset.js

```diff
@@ -1,82 +1,88 @@
+      addForm.bind(this)
+    )
   }
 
-  DynamicFormSet.prototype.addForm = function () {
+  // checks if max forms reached and inserts field before hidden template element
+  const addForm = function () {
     if (this.total < this.maxNum) {
       this.total += 1
-      this.$totalInput.val(this.total)
-      const newForm = getNewForm(this.$formTemplate, this.total - 1)
-      const $newForm = $(newForm).insertBefore(this.$formTemplate)
-      if ($.fn.selectdropdown) {
-        $newForm.find('.category-icon-select').selectdropdown(selectDropdownSettings)
-      }
+      this.totalInput.value
+      const newForm = getNewForm(this.formTemplate, this.total - 1)
+      this.formTemplate.insertAdjacentHTML('beforebegin', newForm)
+  
+      // take new field and add event listener to delete btn
+      this.formTemplate.previousElementSibling.querySelector('.js-remove-form').addEventListener("click",
+        removeForm.bind(this)
+      )
     }
   }
 
-  DynamicFormSet.prototype.removeForm = function (event) {
+  // list all fields and remove field before save
+  const removeForm = function (event) {
     const _this = this
-
     this.total -= 1
-    this.$totalInput.val(this.total)
-
-    const $form = $(event.currentTarget).closest('.js-form')
-    const id = this.$formset.find('.js-form').index($form)
+    this.totalInput.value = this.total
+    const form = event.currentTarget.closest('.js-form')
+    const id = Array.from(this.formset.querySelectorAll('.js-form')).indexOf(form)
 
-    const updateAttr = function ($el, key, i) {
-      if ($el.attr(key)) {
+    // update index prefix of all elements in fields after removed field
+    const updateAttr = function (element, key, i) {
+      // boolean to check attribute
+      if (element.hasAttribute(key)) {
         const _old = _this.prefix + '-' + (id + i + 1)
         const _new = _this.prefix + '-' + (id + i)
-        $el.attr(key, $el.attr(key).replace(_old, _new))
+        element.setAttribute(key, element.getAttribute(key).replace(_old, _new))      
       }
     }
 
-    $form.nextUntil(this.$formTemplate).each(function (i, sibling) {
-      $(sibling).find('*').each(function (j, el) {
-        const $el = $(el)
-        // FIXME: only a limited number of attributes is supported
-        updateAttr($el, 'name', i)
-        updateAttr($el, 'for', i)
-        updateAttr($el, 'id', i)
+    // loop through elements in field and run updateAttr to ensure correct index 
+    let nextSibling = form.nextElementSibling
+    // console.log(form.children)
+    while (nextSibling && nextSibling !== this.formTemplate) {
+      const index = Array.from(this.formset.querySelectorAll('.js-form')).indexOf(nextSibling)
+      const indexTest = index - 1
+      console.log(index)
+      Array.from(nextSibling.firstElementChild.children).forEach(function (element, i) {
+        updateAttr(element, 'name', i)
+        updateAttr(element, 'for', i)
+        updateAttr(element, 'id', i)
```

Comment:
`i` is actually the index of the field child elements, not the index of the parent element, so only the first element has it's attributes replaced as the following index's to not match `_old` variable from `updateAttr`